templates/prometheus-servicemonitor.yaml

```diff
@@ -0,0 +1,31 @@
+{{ if .Values.prometheus.operator.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ template "vault.fullname" . }}
+  labels:
+    helm.sh/chart: {{ include "vault.chart" . }}
+    app.kubernetes.io/name: {{ include "vault.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    {{- toYaml .Values.prometheus.operator.serviceMonitor.selector | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ template "vault.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
```

Comment:
I believe this will be a problem in HA mode as this will select all the pods. From what I understand, only the "active" pod will be able to serve metrics.  I was wondering if it might be good to modify `server-ha-active-service.yaml` to add 1 additional label designating it as active, similar to how the pods do it:
```
apiVersion: v1
kind: Service
metadata:
  name: {{ template "vault.fullname" . }}-active
  namespace: {{ .Release.Namespace }}
  labels:
    helm.sh/chart: {{ include "vault.chart" . }}
    app.kubernetes.io/name: {{ include "vault.name" . }}
    app.kubernetes.io/instance: {{ .Release.Name }}
    app.kubernetes.io/managed-by: {{ .Release.Service }}
    vault-active: true
```

Then the servicemonitor selector can be updated to the following to ensure it only scrapes the active pod in ha mode:
```
  selector:
    matchLabels:
      app.kubernetes.io/name: {{ template "vault.name" . }}
      app.kubernetes.io/instance: {{ .Release.Name }}
      {{- if eq .mode "ha" }}
      vault-active: true
      {{- end }}   
```